File: statusupdate/eventhub.php
<?php
/* Accept event messages and republish them across PGM */
$context = new ZMQContext();

$out = new ZMQSocket($context, ZMQ::SOCKET_PUB);
$out->setSockOpt(ZMQ::SOCKOPT_RATE, 10000);
$out->connect("epgm://eth0;239.192.0.1:7601");

$in = new ZMQSocket($context, ZMQ::SOCKET_PULL);
$in->bind("tcp://*:6767");

$device = new ZMQDevice(ZMQ::DEVICE_FORWARDER, $in, $out);

File: statusupdate/eventhub.erl
-module(eventhub).
-export([run/0]).

run() ->
  {ok, Ctx} = erlzmq:context(),
  {ok, Out} = erlzmq:socket(Ctx, [pub]),
  ok = erlzmq:connect(Out, "epgm://eth0;239.192.0.1:7601"),
  erlzmq:setsockopt(Out, [{rate, 10000}]),
  {ok, In} = erlzmq:socket(Ctx, [pull]),
  ok = erlzmq:bind(In, "tcp://*:6767"),
  loop(In, Out).
  
loop(In, Out) ->
  {ok, Msg} = erlzmq:recv(In),
  ok = erlzmq:send(Out, Msg),
  loop(In, Out).